wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/environment/server/CheEnvironmentEngine.java

```diff
@@ -158,7 +160,12 @@ public CheEnvironmentEngine(SnapshotDao snapshotDao,
                                                 "/recipe/.*$)|(^/recipe/.*$)");
         this.containerNameGenerator = containerNameGenerator;
 
-        eventService.subscribe(new MachineCleaner());
+        eventService.subscribe(event -> {
```

Comment:
It is generally preferable to subscribe outside the constructor e.g.
```java
@PostConstruct
private void subscribe() {
    eventSubscriber.subscribe(...)
}
```
As you implicitly expose instance which is not fully constructed yet